src/common/extensionMessaging.ts

```diff
@@ -0,0 +1,68 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for details.
+
+import * as Q from "q";
+import * as http from "http";
+
+/**
+ * Message server parameters.
+ */
+export let ServerParams = {
+    PORT: 8099,
```

Comment:
The debugger process could try and parse settings.json in the project, but the settings as seen by the extension is a combination of a global settings.json and the project's particular settings.json file, and replicating that logic could be messy.

We could try to have the extension parse the launch.json since we would already require that they all have matching ports (otherwise some configurations would fail) so that might be workable.

We also need to make sure that the server is restartable (since it's registered with the context that might be enough) so that if the settings/launch.json are modified by the user we update the port the server is listening on.